lib/child-writer/index.js

```diff
@@ -113,25 +113,36 @@ exports.write = (image, drive, options) => {
     emitter.emit('error', error);
   };
 
-  ipc.server.on('error', emitError);
-  ipc.server.on('message', (data) => {
-    let message = null;
-
+  /**
+   * @summary Bridge robot message to the child writer caller
+   * @function
+   * @private
+   *
+   * @param {String} message - robot message
+   *
+   * @example
+   * bridgeRobotMessage(robot.buildErrorMessage(new Error('foo')));
+   */
+  const bridgeRobotMessage = (message) => {
     try {
-      message = robot.parseMessage(data);
-    } catch (error) {
-      return emitError(error);
-    }
+      const parsedMessage = robot.parseMessage(message);
+
+      // The error object is decomposed by the CLI for serialisation
+      // purposes. We compose it back to an `Error` here in order
+      // to provide better encapsulation.
+      if (robot.getCommand(parsedMessage) === 'error') {
+        emitError(robot.recomposeErrorMessage(parsedMessage));
```

Comment:
Is doing:
```
const parsedMessage = robot.parseMessage(message);
if (robot.getCommand(parsedMessage) === 'error') {
  emitError(robot.recomposeErrorMessage(parsedMessage));
```
the same as doing:
```
const parsedMessage = robot.parseMessage(message);
if (robot.getCommand(parsedMessage) === 'error') {
  emitError(message);
```
?
(It's kinda hard to keep track of which parts are `String`s, and which parts are `Object`s ;)  )